pkg/collector/autodiscovery/autoconfig.go

```diff
@@ -158,10 +158,11 @@ func (ac *AutoConfig) AddProvider(provider providers.ConfigProvider, shouldPoll
 // Check instances. Should always be run once so providers that don't need
 // polling will be queried at least once
 func (ac *AutoConfig) LoadAndRun() {
-	for _, check := range ac.getAllChecks() {
-		_, err := ac.collector.RunCheck(check)
+	configs := ac.getAllConfigs()
```

Comment:
I think we should kill `getAllChecks()` at this point or put this logic in there.
The problem is we'd have the same issue when `GetChecksByName` is called, that wouldn't return the checks configured with templates.